### ISSUE TITLE
Colored Text Is No Longer Colored

### ISSUE THREAD

**Issue by blackwidowfarms** [Open-Shell/Open-Shell-Menu#512]
**Describe the bug**
White and blue colored text now shows as black, the following code no longer works:

Main2_split_text_color=#FFFFFF,#FFFFFF,#7F7F7F,#7F7F7F
Main2_split_arrow_tint1=#FFFFFF

**To Reproduce**
1: Click on start button
2: See black text on Username, second column text and "Turn Off Computer"

**Expected behavior**
White text on Username (& Turn off Computer)
Blue text on second column entries

**Screenshots**
![Screenshot 2020-10-27 135051](https://user-images.githubusercontent.com/70411948/97361168-683fc680-185c-11eb-897a-83dd9eb0bf91.jpg)

![Screenshot 2020-10-27 135051](https://user-images.githubusercontent.com/70411948/97361213-768de280-185c-11eb-9a5a-5f0f403a70a1.jpg)

**Version:**
 - Open-Shell: 4.4.162
 - OS: Windows 10 v2004 x64

**Additional context**
This doesn't happen with older versions

**Comment by bonzibudd** [comment]
Still works for me. Could you upload the exact skin so we can take a look?

**Comment by bonzibudd** [comment]
Also an xml would be nice.

**Comment by CTVCAM8** [comment]
Possibly text further on down the skin file has overridden it,
or another program (example WindowBlinds) has higher Windows control 
and done the same. 
I can see the taskbar buttons in the picture are not part of Open-Shell.

**Comment by blackcrack** [comment]
screenshoot ?

**Comment by CTVCAM8** [comment]
I'm still using version 4.4.157. I'm sure bonzibudd uses the latest though.

Another couple of possibilities why the Main2 split white text is not showing
is the text is under a different heading in the skin file, 
or the "Split button item" in "Customize Start Menu" has been unchecked so it disregards the white text.
If I remember way back, you wanted to hide the split arrow and shutdown submenu.
Hiding the split arrow is easy "Main2_split_arrow_padding=0,-100,100%" which I think you have done,
but hovering will still open the shutdown submenu. 
If you remove the submenu entries, the text will revert back to dark color again. :(

**Comment by bonzibudd** [comment]
@blackwidowfarms I tested this before in order to verify #433, and I am relatively sure this was an issue before the dll injection was changed. I might be wrong though, so maybe you can try an older version of OSM to see if the text remains the correct color.

**Comment by blackwidowfarms** [comment]
it's all good :) If I roll back a version, then the WB taskbar doesn't get 
skinned but the text is white as it should be, and with current version of OSM 
the text does go back to white once WB is uninstalled





----- Original Message -----

**Comment by blackwidowfarms** [comment]
confirmed text is white on 4.4.161 while WB is loaded





----- Original Message -----

**Comment by ge0rdi** [comment]
> confirmed text is white on 4.4.161 while WB is loaded

There were [minor changes](https://github.com/Open-Shell/Open-Shell-Menu/compare/c88f028f3eb2a0945aa9697d57a0e73098b7c81a...283c0fce03f0804a6d8a79728b2630e588e6dcb2) between 161 and 162.
I can't see how they could affect text color.

Maybe you can try to install 162 from scratch?
- backup your settings
- uninstall 161
- delete `%LOCALAPPDATA%\OpenShell` folder
- delete `HKEY_CURRENT_USER\SOFTWARE\OpenShell` registry key
- install 162

**Comment by blackwidowfarms** [comment]
absolutely, will do that first thing tomorrow (was just shutting down), I will 
follow your instructions exactly and bring you the results





----- Original Message -----

**Comment by blackwidowfarms** [comment]
ok, I uninstalled
- deleted %localappdata%\OpenShell
- deleted reg key
- restarted cold
-Ccleanered my machine and registry to be sure
-installed OSM again

...no change, text is still black in the Username, second column and "turn off computer"
![menu with WB](https://user-images.githubusercontent.com/70411948/97614541-4dde2800-19d7-11eb-9b02-1307ac8c3667.jpg)

- uninstalled WindowBlinds, the text goes back to white 
![no WB](https://user-images.githubusercontent.com/70411948/97614647-6f3f1400-19d7-11eb-8b69-35750e57e21e.jpg)

**Comment by bonzibudd** [comment]
Yeah, that's about what I expected. Could you try the same thing with an older version and see if the issue remains?

**Comment by blackwidowfarms** [comment]
yes sir, same exact steps then with using OSM 4.4.152:
-without WB:
![4 4 152 no WB](https://user-images.githubusercontent.com/70411948/97617282-cd212b00-19da-11eb-8105-f6addf733374.jpg)

-with WB:
![4 4 152 with WB](https://user-images.githubusercontent.com/70411948/97617343-e033fb00-19da-11eb-9c19-f03901561c13.jpg)

**Comment by blackwidowfarms** [comment]
@ge0rdi  my apologies for presuming the recent 161-162 update maybe had something to do with the black text thing. It seems WB-related and I didn't notice that because I wasn't using WB before 162

**Comment by ge0rdi** [comment]
@blackwidowfarms 
No problem. At least it all makes sense now :)

I will eventually try to have a look at it (need to install WB again).

**Comment by CTVCAM8** [comment]
I'm reading all this on the sidelines, not testing anything, busy with other things.
It still has my curiosity for a temporary WB fix though. The free time-limited WB ended for me.

On the screenshot picture, the split text color is black, but the split arrow color stays white which uses tints.
Does that mean if Main2_split_text_color=#FFFFFF,#FFFFFF,#7F7F7F,#7F7F7F had a color mask along with it,
could the tint feature be immune to the change? 

example

Main2_split_text_color=#000000,#000000,#000000,#000000
Main2_split_text_color_mask=#FF0000,#FF0000,#7F0000,#7F0000
Main2_split_text_color_tint1=#FFFFFF

Main2_split_arrow_tint1=#FFFFFF

and for the name color

[USER_NAME]
User_text_color=#FFFFFF
User_text_color_mask=#FF0000
User_text_color_tint1=#FFFFFF

**Comment by blackwidowfarms** [comment]
@CTVCAM8  I'll try that right away

**Comment by blackwidowfarms** [comment]
no change with substituted code lines, whether or not "customize taskbar" is enabled either

**Comment by bonzibudd** [comment]
In the time that I had WindowBlinds installed, I noticed that it had some options to add exceptions to apps. Maybe it would be worth trying to add some exceptions and see if the text color manages to be preserved (Startmenu.exe, even maybe Explorer.exe).

This isn't really ideal as a proper solution, but it might help and see if WindowBlinds is inherently breaking something that Open-Shell uses, or if it's just a weird glitch that the combination of the programs exposes.

**Comment by blackwidowfarms** [comment]
I just tried adding exceptions to Explorer.exe and startmenu.exe together, the text styling (white and blue) plus highlighting came back! but explorer windows aren't skinned.
 I tried only startmenu.exe as a WB exception but it's back to black text.
Is it somehow WBs priority with explorer.exe over startmenu.exe ?

**Comment by bonzibudd** [comment]
@blackwidowfarms not necessarily priority, but it might be because Open-Shell heavily relies on injection into the Explorer process, and if WindowBlinds overrides text in Explorer, the same effects could show up in Open-Shell. I'm not a programmer though, so @ge0rdi would probably know better about that process.

**Comment by ge0rdi** [comment]
`Open-Shell` is loaded inside Explorer (`StartMenuDLL.dll`). So it cannot be excluded this way.

**Comment by MisterMonopoly** [comment]
Phew, first time commenting on a discussion issue here on github, but I know how to fix this, it just requires to modify the eXPerience skin in skinstudio, found out about this yesterday

1. you need to download and install skinstudio

2. now you need to open the eXPerience skin and select the substyle you want to modify

3. now with all of that selected, go to the controls tab and then modify every single control config in there, specifically where the text tab is located, for some reason the color is forced into black, and that reproduces the issue, you just need to put every single control color into the OS default and with that, you can solve the problem

Now you can enjoy the immertion of the eXPerience once again, cheers!

**Comment by DJJolty257** [comment]
@MisterMonopoly I wanted to know how to pull this off since I can't find anything related to the colour that OpenShell is using for the start menu. Tried changing all of the text-based values to white and it did absolutely nothing.

**Comment by MisterMonopoly** [comment]
> @MisterMonopoly I wanted to know how to pull this off since I can't find anything related to the colour that OpenShell is using for the start menu. Tried changing all of the text-based values to white and it did absolutely nothing.

On skinstudio or on the open shell configs? hmm weird, on the controls tab i've put everything to OS default and that resolved my issue, specially with the silver and olive themes. Give me details and I'll try to research what's going on :P

**Comment by 2kkenan** [comment]
yo we still need a fix for this, ive changed all colour values in skin studio - didnt work, ive created exceptions for open shell - didnt work, i have reinstalled both openshell and windows blinds multiple times - still didnt work. 
I noticed that using a different skin brought back the white text, but the one i want to use still has the black text (eXperience11)....
Can we try find a fix for this?

**Comment by r5400** [comment]
As of August 2025, If you are using eXperience11 and having the issue where the start menu text appears black:

Open the skin in Stardock SkinStudio, go to Edit Controls -> Edit Toolbar Controls -> Button (6 State) and set Text light/dark to OS Default. Then save it and apply again the skin in windowblinds.

**Comment by DNHND28** [comment]
> yo we still need a fix for this, ive changed all colour values in skin studio - didnt work, ive created exceptions for open shell - didnt work, i have reinstalled both openshell and windows blinds multiple times - still didnt work. I noticed that using a different skin brought back the white text, but the one i want to use still has the black text (eXperience11).... Can we try find a fix for this?

2026 here, same skin same issue lol

**Comment by CTVCAM8** [comment]
One person got it working with Open-Shell 4.4.195

 [#2397](https://github.com/Open-Shell/Open-Shell-Menu/discussions/2397)

I haven't used WindowBlinds for years.
Does Stardock  Start10/11 menu software not work anymore with WindowBlinds skins?

**Comment by DNHND28** [comment]
> One person got it working with Open-Shell 4.4.195
> 
> [#2397](https://github.com/Open-Shell/Open-Shell-Menu/discussions/2397)
> 
> I haven't used WindowBlinds for years. Does Stardock Start10/11 menu software not work anymore with WindowBlinds skins?

start11 still works, but for people who'd prefer open-shell, use skinstudio:

https://www.reddit.com/r/Stardock/comments/1k8dcu8/username_on_start_menu_is_always_black_how_do_i/ 

-there's a solution under sternjaeger's comment, but also edit the normal Button tab and not just the (6 state) variant; this works for me, even on 4.4.196
